content/docs/ai/cloud-providers.md

```diff
@@ -189,6 +189,226 @@ To set up OpenAI, continue with the [Authenticate to the LLM](/docs/ai/auth/) gu
+
+### Configuration Options
+
+**Path Override**
+- `pathOverride.fullPath` - Specifies a custom API path to replace the provider's default path
+- Supported for OpenAI and Anthropic compatible APIs
+- **Example use case**: Azure OpenAI requires deployment-specific paths like `/openai/deployments/{deployment-name}/chat/completions`
+- **Note**: The full path should include any query parameters required by your custom endpoint
+
+**Auth Header Override** 
+- `authHeaderOverride.headerName` - Custom header name (default: "Authorization")
+- `authHeaderOverride.prefix` - Custom prefix (default: "Bearer ")
+- **Important**: When using no prefix, set `prefix: ""` (empty string)
+
+### Common Use Cases
+
+**Path Override Examples:**
+- **Custom API Gateways**: When routing through your own API gateway that uses different paths
+- **API Versioning**: When you need to use a specific API version not supported by default
+- **Proxy Services**: When using a proxy service that modifies the standard endpoint structure
+- **Azure OpenAI**: Requires deployment-specific paths with API version parameters
+
+**Auth Header Override Examples:**
+- **Azure OpenAI**: Uses `api-key` header without `Bearer` prefix
+- **Custom Authentication**: When your organization requires specific header formats
+- **API Proxies**: When routing through services that expect different auth formats
+
+### Practical Examples
+
+#### Example 1: Custom API Endpoint (Path Override)
+
+This example shows how to configure a custom API endpoint path for an OpenAI-compatible provider:
+
+```yaml
+apiVersion: gateway.kgateway.dev/v1alpha1
+kind: Backend
+metadata:
+  name: custom-openai
+  namespace: {{< reuse "docs/snippets/namespace.md" >}}
+  labels:
+    app: ai-gateway
+spec:
+  ai:
+    llm:
+      provider:
+        openai:
+          model: gpt-4
+          authToken:
+            kind: SecretRef
+            secretRef:
+              name: openai-secret
+          pathOverride:
+            fullPath: "/v2/custom/chat/completions"  # Custom endpoint path
+  type: AI
+```
+
+#### Example 2: Custom Authentication (Auth Header Override)
+
+This example demonstrates how to configure custom authentication headers:
+
+```yaml
+apiVersion: gateway.kgateway.dev/v1alpha1
+kind: Backend
+metadata:
+  name: custom-auth-provider
+  namespace: {{< reuse "docs/snippets/namespace.md" >}}
+  labels:
+    app: ai-gateway
+spec:
+  ai:
+    llm:
+      provider:
+        openai:
+          model: gpt-4
+          authToken:
+            kind: SecretRef
+            secretRef:
+              name: custom-auth-secret
+          authHeaderOverride:
+            headerName: "X-API-Key"       # Custom header name
+            prefix: "Token "              # Custom prefix
+  type: AI
+```
+
+### Complete Example: Azure OpenAI Setup
```

Comment:
I think we can remove all of this, as the API docs as well as the Azure example that you provide covers all of this.
```suggestion

```